Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -409,8 +409,7 @@ open class RouteController: NSObject {
         updateSpokenInstructionProgress(status: status, willReRoute: isRerouting)
         updateVisualInstructionProgress(status: status)
         updateRoadName(status: status)
-        routeProgress.updateDistanceToIntersection(from: snappedLocation)
```

Comment:
It looks like this pr fixes one of the possible reasons for #4314

Before this pull request, `intersectionsIncludingUpcomingManeuverIntersection` was updated too late. `intersectionIndex` was changed earlier in the code.  And there was a moment of using an old `intersectionsIncludingUpcomingManeuverIntersection` together with the new value of `intersectionIndex`.

----

1. A new value for `intersectionIndex` was set in the `RouteController.updateIndexes(status:progress: )`
2. And in the `RouteController.update(progress:status:with:rawLocation:upcomingRouteAlerts:mapMatchingResult:routeShapeIndex:)`  the new `intersectionIndex` was already used with an old `intersectionsIncludingUpcomingManeuverIntersection` value
3. only after that the `intersectionsIncludingUpcomingManeuverIntersection` value was updated 

This could have led to the incorrect data.

Reply:
There were no problems with other properties here because for the voice instructions and the display instructions we only update the index, but for  intersections we have this settable property `intersectionsIncludingUpcomingManeuverIntersection`, that should be in sync with `intersectionIndex`